homeassistant/components/fritz/__init__.py

```diff
@@ -1 +1,87 @@
-"""The fritz component."""
+"""Support for AVM Fritz!Box functions."""
+import logging
+
+from fritzconnection.core.exceptions import FritzConnectionException
+
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_REAUTH, ConfigEntry
+from homeassistant.const import (
+    CONF_DEVICES,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+)
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .common import FritzBoxTools
+from .const import DEFAULT_HOST, DEFAULT_PORT, DOMAIN, SUPPORTED_DOMAINS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up FRITZ!Box Tools component."""
+    if DOMAIN not in config:
+        return True
+
+    for entry_config in config[DOMAIN][CONF_DEVICES]:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=entry_config
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up fritzboxtools from config entry."""
+    _LOGGER.debug("Setting up FRITZ!Box Tools component")
+    host = entry.data.get(CONF_HOST, DEFAULT_HOST)
+    port = entry.data.get(CONF_PORT, DEFAULT_PORT)
+    username = entry.data.get(CONF_USERNAME)
+    password = entry.data.get(CONF_PASSWORD)
+
+    fritz_tools = FritzBoxTools(
+        hass=hass,
+        host=host,
+        port=port,
+        username=username,
+        password=password,
+    )
+
+    try:
+        await fritz_tools.async_setup()
+    except FritzConnectionException:
+        _LOGGER.error("Unable to setup FRITZ!Box Tools component")
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_REAUTH},
+                data=entry,
+            )
+        )
+        return False
```

Comment:
```suggestion
    except FritzConnectionException as ex:
        raise ConfigEntryAuthFailed from ex
```

Reply:
We should raise `ConfigEntryNotReady` here also if its just offline so it retries later.

Reply:
I cleaned up as suggested and also added the `ConfigEntryNotReady`